File: src/main/kotlin/com/munyu/wamis/domain/Customer.kt
package com.munyu.wamis.domain

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.util.ArrayList
import javax.persistence.*

@Entity
data class Customer(
        @Column(nullable = false)
        val firstName: String,

        @Column(nullable = true)
        val middleName: String,

        @Column(nullable = false)
        val surname: String,

        @Column(nullable = false)
        val idNumber: String,

        @Column(nullable = true)
        val email: String,

        @Column(nullable = true)
        @OneToMany(cascade = [CascadeType.ALL])
        val phoneNumbers: MutableList<PhoneNumber>? = null,

        @JsonIgnoreProperties("customer")
        @OneToMany(mappedBy = "customer",cascade= [CascadeType.ALL])
        val connections: MutableList<Connection> = ArrayList()

) : AbstractAuditableEntity() {

    fun addConnection(connections: List<Connection>){
        this.connections.addAll(connections);
        connections.forEach { c -> c.customer = this }
    }
}

File: src/main/webapp/src/app/model/customer.ts
import {Connection} from './connection';
import {PhoneNumber} from './phone.number';

export class Customer {
  id: number;
  firstName: string;
  middleName: string;
  surname: string;
  idNumber: string;
  email: string;
  phoneNumbers: PhoneNumber[];
  connections: Connection[];
}

export interface CustomerApi {
  items: Customer[];
  total_count: number;
}
